### PR TITLE
8282690: runtime/CommandLine/VMDeprecatedOptions.java fails after JDK-8281181

### PR DESCRIPTION
**Pull request by iklam**
Please review this small fix. The two options `UseContainerCpuShares` and `PreferContainerQuotaForCPUCount` are available only on Linux, so they are tested under `if (Platform.isLinux()) { ...}`

Tested with tiers 1-4

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282690](https://bugs.openjdk.java.net/browse/JDK-8282690): runtime/CommandLine/VMDeprecatedOptions.java fails after JDK-8281181

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7712/head:pull/7712` \
`$ git checkout pull/7712`

Update a local copy of the PR: \
`$ git checkout pull/7712` \
`$ git pull https://git.openjdk.java.net/jdk pull/7712/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7712`

View PR using the GUI difftool: \
`$ git pr show -t 7712`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7712.diff">https://git.openjdk.java.net/jdk/pull/7712.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java [hunk 1]
```diff
@@ -54,15 +55,22 @@
...
-            {"PreferContainerQuotaForCPUCount", "true"},
 
             // deprecated alias flags (see also aliased_jvm_flags):
             {"DefaultMaxRAMFraction", "4"},
             {"CreateMinidumpOnCrash", "false"}
           }
         ));
         DEPRECATED_OPTIONS = deprecated.toArray(new String[][]{});
+
+        ArrayList<String[]> deprecated_linux = new ArrayList(
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java:66]: Seems simpler to just add these to `deprecated` before line 64.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java (hunk 1 above)
Comment (dcubed-ojdk) [reply test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java:66]: So move the `Platform.isLinux()` check up here and optionally
ArrayList.add() the needed strings? That does sound simpler.

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java (hunk 1 above)
Comment (iklam) [reply test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java]: I changed the code as David suggested.

I also added a comment to explain why we use the clumsy-looking  `new ArrayList(Arrays.asList(new String[][] {...}))` construct. A simple `new String[][] {...}` would suffice if we didn't need to add optional checks for specific platforms.

**Review by dholmes-ora — APPROVED** [review]
Looks good - thanks.

IIRC we had more complex usage in the past which is why we still have the seemingly complex setup.

**Comment by iklam** [comment]
Tested tiers 1-2 again. Verified that VMDeprecatedOptions.java succeeded on Linux/macOS/Windows, and the affected options are used only on Linux.

Thanks @dcubed-ojdk and @dholmes-ora for the review.

/integrate

**Pull request closed without merging**